crates/tabby-common/src/config.rs

```diff
@@ -128,8 +129,17 @@ impl RepositoryConfig {
             .unwrap_or_else(|_| url.to_string())
     }
 
-    pub fn dir(&self) -> PathBuf {
-        Self::resolve_dir(&self.git_url)
+    pub fn dir(&self, source_id: &str) -> PathBuf {
+        Self::get_dir(&self.git_url, source_id)
+    }
+
+    pub fn get_dir(git_url: &str, source_id: &str) -> PathBuf {
+        if Self::resolve_is_local_dir(git_url) {
+            let path = git_url.strip_prefix("file://").unwrap();
+            path.into()
+        } else {
+            repositories_dir().join(Self::resolve_dir_name(source_id))
+        }
     }
```

Comment:
can this be merged to `resolve_dir`?

e.g `pub fn resolve_dir(source_id: &str, git_url: Option<&str>)`

Reply:
Yes. We can merge them to one method